sklearn/linear_model/_ridge.py

```diff
@@ -595,6 +613,8 @@ def _ridge_regression(
         elif return_intercept:
             # sag supports fitting intercept directly
             solver = "sag"
+        elif is_array_api_compliant:
+            solver = "svd"
```

Comment:
I am split on this. "auto" is practical, and I think the added opacity here is marginal, it is already is very opaque... From the look of it the "auto" heuristics are not currently described in the documentation, and it is also not possible to inspect the chosen solver after the fit, since there is no `.solver_`-like attribute. (well that is reasonably cheap to fix, but it has not been a concern so far) .

On the other hand, ultimately I think we shouldn't have "auto" for now in this PR, not because of opacity, but because there are cases where svd on gpu does not actually beat cholesky.lsqr/sparse_cg on cpu. But if we plan to support one of cholesky/lsqr/sparse_cg then i think we should add "auto" .

Reply:
numpy-based tests keep passing without this instruction so I suspect the issue was with other libraries that print `array(1)` rather than `1` in some tests, I presume it can also be dealt with by using `float` rather than `reshape` + `getitem`. Will fix.

Reply:
We talked about this a bit today (Olivier, Jerome and Tim) and a solution we could agree on is to use `auto` for Array API input and select `svd` for the user but have a warning that informs the user about this. The warning text should contain the solver that would have been chosen with a Numpy array as input as well as the solver that was actually chosen.

Reply:
Something to keep in mind is that if a user passes a Numpy array (with array api dispatch turned off) and when they pass a Numpy array (with dispatch on) the same thing should happen. The reason we need to explicitly think about this case is that with dispatching on a Numpy array will be wrapped and be treated as "Array API input".

Reply:
We might also want to store the effectively chosen solver as a fitted attribute:

```python
    self.solver_ = "svd"
```

This will ease debugging in case of ambiguity because of complex logic for the `solver="auto"` case.

Please also add a dedicated test for the error message raised when passing a non-numpy Array API input with the default "auto" solver and `positive=True`.

Reply:
Very much agree, I will update accordingly thanks.

Reply:
done